File: typescript/src/n0053_maximum_subarray.ts
export function maxSubarray(numbers: number[]): number {
  let max = numbers[0]!;
  let sum = numbers[0]!;

  for (const number of numbers.slice(1)) {
    if (sum < 0) {
      sum = 0;
    }
    sum += number;
    max = Math.max(sum, max);
  }

  return max;
}

File: rust/src/n0053_maximum_subarray.rs
pub fn max_subarray(numbers: &[i32]) -> i32 {
    let mut max = numbers[0];
    let mut sum = numbers[0];

    for &number in &numbers[1..] {
        if sum < 0 {
            sum = 0;
        }
        sum += number;
        max = max.max(sum);
    }

    max
}
